### ISSUE TITLE
Output returns pdf "code" to browser but browser does not render pdf

### ISSUE THREAD

**Issue by schmidtl4** [spipu/html2pdf#367]
Hello I have successfully installed html2pdf and can successfully write a pdf to file.  I am unable, however to have a pdf appear in the latest versions of Chrome and Firefox.  If I look at developer tools I see the code for the pdf being returned, but nothing new displays in the browser.

Two questions:
1) is the normal expected behavior of $html2pdf->output(); to the same tab/windows that called it or to a new tab?
2) What troubleshooting steps can be taken to identify whether the reason the pdf is not displaying is a browser issue or  my issue or the result of a html2pdf bug?  

I've tried uninstalling/reinstalling browsers. 
I've tried uninstalling/reinstalling Adobe Reader DC.  I've set the default reader (in the Windows setings) for pdfs to (alternately): Adobe Reader DC, Adobe Acrobat (10), and directly to Chrome.  The pdf just does not display in the browser.

Would appreciate helping me identify the source of this mystery. 

Thank you!

**Comment by LittleBigFox** [comment]
Hi!
The default is "send to the browser" (https://github.com/spipu/html2pdf/blob/master/doc/output.md). So it's in the current tab.

However, you can try to open your result within a new window (JS window.open), it may help you? Or try to force download with 'D' option, for testing your PDF?

Regards,
LBF

**Comment by schmidtl4** [comment]
Thanks for the response.

Unfortunately, the only output that works is to save to file ('F').  
I've tried

$html2pdf->output(); //encoded string returned to browser, but no display
$html2pdf->output($filepath, 'F'); //works
$html2pdf->output($filepath, 'D'); // encoded string returned to 
browser, but no prompt for download
$html2pdf->output($filepath, 'I'); // encoded string returned to 
browser, but no display

I've tried this in Firefox and Chrome - same result.

In developer tools, there is a 200 response to the pdf request, but 
nothing is triggered.

I've turned popup blockers off on both browsers and checked (in Firefox) 
that the PDF extension is set to "preview in Firefox". No change.

I've disabled all extensions and plugins in Firefox and still no joy.

Any other suggestions?

Thanks,
Les

**Comment by LittleBigFox** [comment]
Hummm... 
- You can try to put `header("Content-type:application/pdf");` in start of your php file...
- Or maybe have you a warning or something like, have you got some error.log in your php server? It may break the PDF generation...
- It may be a bug with ob_start, ob_end_flush, etc, try to look at some example in the exemple directory?
- When you used the 'D' option, can you open the PDF file?

Regards,
LBF

**Comment by schmidtl4** [comment]
Thanks for your continued suggestions.
None of the items resolved my existing code.
I've now built a hello world "application" that uses the same core pdf 
generation code I was using before and I have a basic pdf working.  
Which means that something in my code was causing it to error out with 
no error messages.  Still digging into that.

I am having an issue with page numbers though.  When I add 
"footer='page'" to the <page> tag all I get is ##### at the bottom.  See 
attached.  I'm running version 5.1.

Thanks,
Les

**Comment by spipu** [comment]
did you try just  to test directly the examples ? to see if the pb is coming from you app or html2pdf ?

**Comment by NejiMehrez** [comment]
Hello I am having the same problem, I used to create a PDF file and open it in an iframe it was working perfectly BUT now it is not working anymore only in CHrome it is working in all other web browser, and even the regular HTML2PDF samples they don't work in Chrome i t means it is not a problem in my page. It is making me crazy I tried everything  I know and it is still not working.
sing output with 'D' is working fine even in chrome but using Output with 'I' or with no parameters doesn't work

PLEASE HELP

Thank you

**Comment by LittleBigFox** [comment]
Hi!

Have you change previously the content-type header? 
header("Content-Type...") 
Or have you some error in your logs?

Regards,
LBF

**Comment by NejiMehrez** [comment]
Hi

 

Thank you for your reply I find the cause of the problem, I recently did some changes in the service-worker.js file in the
self.addEventListener('fetch', function(event) {} function and that was causing the issue.

I disabled that function until i find why it is causing that.

I will close the issue

 

Thanks again

 

 

De : LittleBigFox [mailto:notifications@github.com] 
Envoyé : mercredi 9 janvier 2019 09:47
À : spipu/html2pdf
Cc : NejiMehrez; Comment
Objet : Re: [spipu/html2pdf] Output returns pdf "code" to browser but browser does not render pdf (#367)

 

Hi!

Have you change previously the content-type header?
header("Content-Type...")
Or have you some error in your logs?

Regards,
LBF
